suzieq/shared/sqPlugin.py

```diff
@@ -0,0 +1,66 @@
+import abc
+from importlib import import_module
+from importlib.util import find_spec
+from inspect import getmembers, getmro, isclass
+from pkgutil import iter_modules
+from typing import Dict, Type
+
+
+class SqPlugin:
+    """SqPlugin is the base common class inherited by all the
+    Suzieq plugins
+    """
+
+    @abc.abstractmethod
+    def __init__(self) -> None:
+        pass
+
+    @classmethod
+    def discover_plugins(cls,
```

Comment:
I don't like the name discover_plugins that returns a dict of plugins. Is a name such as get_plugins better OR we can have discover_plugins which updates a class variable called plugins?

Reply:
I think they are both equally good, Probably, only renaming to `get_plugins()` should be enough, since plugin discovery is typically a one-shot operation we might not need to store the result. However, if you prefer the class variable is completely fine for me